Tasks/Gradle/sonarqube-common/sonarqube-server.ts

```diff
@@ -0,0 +1,112 @@
+import Q = require('q');
+import url = require('url');
+import https = require('https');
+import http = require('http');
+import {IncomingMessage} from 'http';
+
+import {SonarQubeEndpoint} from './sonarqube-common';
+import sqCommon = require('./sonarqube-common');
+
+import tl = require('vsts-task-lib/task');
+
+export interface ISonarQubeServer {
+    callSonarQubeRestEndpoint(path:string):Q.Promise<Object>;
+
+    getEndpoint():SonarQubeEndpoint;
+}
+
+export class SonarQubeServer implements ISonarQubeServer {
+
+    private endpoint:SonarQubeEndpoint;
```

Comment:
Could you move all the endpoint logic to a separate file? We really need to clean up that sonarqube-common.ts file http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule